### PR TITLE
[Debt] Laravel11 upgrade

### LINKED ISSUE #9754: ♻️ Upgrade to Laravel 11
## ♻️ Debt/Refactor

Laravel 11 is out!
https://laravel-news.com/laravel-11

## 🙋‍♀️ Proposed Solution

https://laravel.com/docs/11.x/upgrade#upgrade-11.0

## ✅ Acceptance Criteria

- [ ] API is updated to version 11

## 🛑 Blockers

```[tasklist]
### Blocked By
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10187
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/11310
```

### PR DESCRIPTION
**Pull request by brindasasi**
🤖 Resolves #9754 

## 👋 Introduction

Upgrade to Laravel 11 

## 🧪 Testing

1. Run Composer Update
2. Verify all tests pass 
3. Verify application behaves normally without any issues

### CODE REVIEW

**Comment by brindasasi** [comment]
There are no errors in local. Any idea why its failing on CI @esizer  ? 
Screenshot 2024-10-15 at 10 19 48 AM

**Comment by esizer** [comment]
> There are no errors in local. Any idea why its failing on CI @esizer ?
 
Hmm, I am getting the same errors locally as well. Well, not errors but warnings about ignores that are no longer needed.

> ```shell
>  360/360 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
> 
>  ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>   Line   Http/Controllers/AuthController.php
>  ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>          Ignored error pattern #^Call to an undefined method Illuminate\\Http\\Client\\Response\:\:getBody\(\)\.$# in path /var/www/html/api/app/Http/Controllers/AuthController.php was not
>          matched in reported errors.
>  ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> 
>  ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>   Line   Http/Controllers/SupportController.php
>  ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>          Ignored error pattern #^Call to an undefined method Illuminate\\Http\\Client\\Response\:\:getBody\(\)\.$# in path /var/www/html/api/app/Http/Controllers/SupportController.php was
>          not matched in reported errors.
>  ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> 
>  ------ -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>   Line   Services/OpenIdBearerTokenService.php
>  ------ -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>          Ignored error pattern #^Call to an undefined method Illuminate\\Http\\Client\\Response\:\:getBody\(\)\.$# in path /var/www/html/api/app/Services/OpenIdBearerTokenService.php was
>          not matched in reported errors.
>  ------ -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> 
> 
>  [ERROR] Found 3 errors
> ```

**Comment by brindasasi** [comment]
> > There are no errors in local. Any idea why its failing on CI @esizer ?
> 
> Hmm, I am getting the same errors locally as well. Well, not errors but warnings about ignores that are no longer needed.
> 
> > ```shell
> >  360/360 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
> > 
> >  ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> >   Line   Http/Controllers/AuthController.php
> >  ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> >          Ignored error pattern #^Call to an undefined method Illuminate\\Http\\Client\\Response\:\:getBody\(\)\.$# in path /var/www/html/api/app/Http/Controllers/AuthController.php was not
> >          matched in reported errors.
> >  ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> > 
> >  ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> >   Line   Http/Controllers/SupportController.php
> >  ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> >          Ignored error pattern #^Call to an undefined method Illuminate\\Http\\Client\\Response\:\:getBody\(\)\.$# in path /var/www/html/api/app/Http/Controllers/SupportController.php was
> >          not matched in reported errors.
> >  ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> > 
> >  ------ -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> >   Line   Services/OpenIdBearerTokenService.php
> >  ------ -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> >          Ignored error pattern #^Call to an undefined method Illuminate\\Http\\Client\\Response\:\:getBody\(\)\.$# in path /var/www/html/api/app/Services/OpenIdBearerTokenService.php was
> >          not matched in reported errors.
> >  ------ -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> > 
> > 
> >  [ERROR] Found 3 errors
> > ```

 If I remove my local doesn't like it 😆 .. probably some cache .. let me try again

File: api/tests/Unit/AcceptLanguageMiddlewareTest.php [hunk 1]
```diff
@@ -42,8 +42,8 @@ public static function provider(): array
             'fallback to English when not set' => ['unset', 'en'],
             // Can get an expected locale out of a list of unexpected values
             'sets English from list' => ['mt-MT;q=0.9,en-US;q=0.8,eu-ES;q=0.7', 'en'],
-            'sets French from list' => ['mt-MT;q=0.9,fr-LU;q=0.8,eu-ES;q=0.7', 'fr'],
-            'weights by quality' => ['en;q=0.8,eu-ES;q=0.7,fr-LU;q=0.9;', 'fr'],
+            'sets French from list' => ['mt-MT;q=0.9,fr;q=0.8,eu-ES;q=0.7', 'fr'],
```
Comment (brindasasi) [inline api/tests/Unit/AcceptLanguageMiddlewareTest.php:45]: fr-LU is no longer working in getPreferredMethod.

**Review by esizer — APPROVED** [review]
LGTM! :+1:
